In [16]:
# %load_ext autoreload
%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [1]:
import sys
import preprocess
import pickle
import numpy as np

1. Books
2. Electronics
3. Movies and TV
4. CDs and Vinyl
5. Clothing, Shoes and Jewelry
6. Home and Kitchen
7. Kindle Store
8. Sports and Outdoors
9. Cell Phones and Accessories
10. Health and Personal Care
11. Toys and Games
12. Video Games
13. Tools and Home Improvement
14. Beauty
15. Apps for Android
16. Office Products
17. Pet Supplies
18. Automotive
19. Grocery and Gourmet Food
20. Patio, Lawn and Garden
21. Baby
22. Digital Music
23. Musical Instruments
24. Amazon Instant Video

In [2]:
# get all the userID and itemID
# it has already be processed
for domainIdx in range(5,25):
    domainMetaName = preprocess.loadMetaDomain(domainIdx)
    # print(domainMetaName)
    ratingFileName = 'data/amazon/ratings_' + domainMetaName+'.csv'
    # print(ratingFileName)
    preprocess.getAllUserItems(ratingFileName)

data/amazon/ratings_Clothing_Shoes_and_Jewelry total Statistic (user, item, ratings): 3117268 1136004 5748920
data/amazon/ratings_Home_and_Kitchen total Statistic (user, item, ratings): 2511610 410243 4253926
data/amazon/ratings_Kindle_Store total Statistic (user, item, ratings): 1406890 430530 3205467
data/amazon/ratings_Sports_and_Outdoors total Statistic (user, item, ratings): 1990521 478898 3268695
data/amazon/ratings_Cell_Phones_and_Accessories total Statistic (user, item, ratings): 2261045 319678 3447249
data/amazon/ratings_Health_and_Personal_Care total Statistic (user, item, ratings): 1851132 252331 2982326
data/amazon/ratings_Toys_and_Games total Statistic (user, item, ratings): 1342911 327698 2252771
data/amazon/ratings_Video_Games total Statistic (user, item, ratings): 826767 50210 1324753
data/amazon/ratings_Tools_and_Home_Improvement total Statistic (user, item, ratings): 1212468 260659 1926047
data/amazon/ratings_Beauty total Statistic (user, item, ratings): 1210271 24927

In [31]:
# transforming the amazon training data to the baseline's required dataset form
domainIdx = 17
domainMetaName = preprocess.loadMetaDomain(domainIdx)
metaName = 'data/amazon/ratings_' + domainMetaName
print(metaName)
# trainData, testData = preprocess.loadTrainTest(metaName)
preprocess.transformDataSet(metaName)

data/amazon/ratings_Pet_Supplies
-----transforming training file----------
-----transforming testing file----------
-----transforming negative file----------


In [ ]:
dataInfoPath = 'data/amazon/domain_info.txt'
with open(dataInfoPath, 'r') as df:
    for line in df:
        numU = int(line.split()[-3])
        numI = int(line.split()[-2])
        numRating = float(line.split()[-1])
        sparsity = (numRating+0.0)/(numU * numI)
        print(line.split('(')[0]+ ':' + str(sparsity))
#         print(numU)

In [ ]:
#get the number of common users
for domainIdx in range(1,25):
    for domainIdx_2 in range(1,25):
        if domainIdx != domainIdx_2:
            domainMetaName_1 = preprocess.loadMetaDomain(domainIdx)
            domainMetaName_2 = preprocess.loadMetaDomain(domainIdx_2)
        # print(domainMetaName)
            allUserMetaName_1 = 'data/amazon/ratings_' + domainMetaName_1
            allUserMetaName_2 = 'data/amazon/ratings_' + domainMetaName_2
            users = preprocess.allCommonUser(allUserMetaName_1,allUserMetaName_2)
#     print(len(users))

In [4]:
# ratingFileName1 = 'data/amazon/ratings_Amazon_Instant_Video.csv'
# ratingFileName2 = 'data/amazon/ratings_Video_Games.csv'
# ratingFileName3 = 'data/amazon/ratings_Books.csv'
# ratingFileName4 = 'data/amazon/ratings_Movies_and_TV.csv'
# ratingFileName5 = 'data/amazon/ratings_Tools_and_Home_Improvement.csv'
# ratingFileName6 = 'data/amazon/ratings_Home_and_Kitchen.csv'
# ratingFileName7 = 'data/amazon/ratings_Automotive.csv'
# ratingFileName8 = 'data/amazon/ratings_Digital_Music.csv'
# ratingFileName9 = 'data/amazon/ratings_' + preprocess.loadMetaDomain(9) + '.csv'
# ratingFileName10 = 'data/amazon/ratings_' + preprocess.loadMetaDomain(10) + '.csv'
# ratingFileName17 = 'data/amazon/ratings_Pet_Supplies.csv'

# metaName1 = ratingFileName1.split('.')[0]
# metaName2 = ratingFileName2.split('.')[0]
# metaName3 = ratingFileName3.split('.')[0]
# metaName4 = ratingFileName4.split('.')[0]
# metaName5 = ratingFileName5.split('.')[0]
# metaName6 = ratingFileName6.split('.')[0]
# metaName7 = ratingFileName7.split('.')[0]
# metaName8 = ratingFileName8.split('.')[0]

# print(metaName1)
# print(metaName2)
# print(metaName3)
# print(metaName4)
# print(metaName5)
# print(metaName6)
# print(metaName7)
# print(metaName8)
ratingFileName = [0]
metaName = [0]
for metaNameIdx in range(1,25):
    tempFileName = 'data/amazon/ratings_' + preprocess.loadMetaDomain(metaNameIdx) + '.csv'
    print(tempFileName)
    ratingFileName.append(tempFileName)
    tempMetaName = tempFileName.split('.')[0]
    print(tempMetaName)
    metaName.append(tempMetaName)

data/amazon/ratings_Books.csv
data/amazon/ratings_Books
data/amazon/ratings_Electronics.csv
data/amazon/ratings_Electronics
data/amazon/ratings_Movies_and_TV.csv
data/amazon/ratings_Movies_and_TV
data/amazon/ratings_CDs_and_Vinyl.csv
data/amazon/ratings_CDs_and_Vinyl
data/amazon/ratings_Clothing_Shoes_and_Jewelry.csv
data/amazon/ratings_Clothing_Shoes_and_Jewelry
data/amazon/ratings_Home_and_Kitchen.csv
data/amazon/ratings_Home_and_Kitchen
data/amazon/ratings_Kindle_Store.csv
data/amazon/ratings_Kindle_Store
data/amazon/ratings_Sports_and_Outdoors.csv
data/amazon/ratings_Sports_and_Outdoors
data/amazon/ratings_Cell_Phones_and_Accessories.csv
data/amazon/ratings_Cell_Phones_and_Accessories
data/amazon/ratings_Health_and_Personal_Care.csv
data/amazon/ratings_Health_and_Personal_Care
data/amazon/ratings_Toys_and_Games.csv
data/amazon/ratings_Toys_and_Games
data/amazon/ratings_Video_Games.csv
data/amazon/ratings_Video_Games
data/amazon/ratings_Tools_and_Home_Improvement.csv
data/amazon/rat

In [30]:
preprocess.processing(ratingFileName1, num_items = 5, train_items = 10, num_users = 0)
# preprocess.processing(ratingFileName2,num_items = 5, train_items = 10, num_users = 3)
# preprocess.processing(ratingFileName3,num_items = 30, train_items = 50, num_users = 30)
# preprocess.processing(ratingFileName4, num_items = 15, train_items = 20, num_users = 10)
# preprocess.processing(ratingFileName5,num_items = 5, train_items = 10, num_users = 5)
# preprocess.processing(ratingFileName6,num_items = 10, train_items = 15, num_users = 5)
# preprocess.processing(ratingFileName7,40)
# preprocess.processing(ratingFileName8,60)
# preprocess.processing(ratingFileName[17],num_items = 10, train_items = 15, num_users = 5)

total Statistic: 740985 103288 1235316
starting dump the id of users and items
Remaning: 488 1812
0.007393786414793906


In [64]:
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[1],percent=1.0, 
#                                                    train_items=2,user_thres = 2,item_thres=1)
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[2],percent=1.0, 
#                                                    train_items=2,user_thres = 2,item_thres=1)
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[3],percent=1.0, 
#                                                    train_items=5,user_thres = 3,item_thres=1)
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[4],percent=1.0, 
#                                                    train_items=2,user_thres = 2,item_thres=1)
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[5],percent=1.0, 
#                                                    train_items=1,user_thres = 2,item_thres=1)
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[6],percent=1.0, 
#                                                    train_items=1,user_thres = 2,item_thres=1)
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[7],percent=1.0, 
#                                                    train_items=1,user_thres = 2,item_thres=1)
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[8],percent=1.0, 
#                                                    train_items=1,user_thres = 1,item_thres=1)
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[9],percent=1.0, 
#                                                    train_items=1,user_thres = 1,item_thres=1)
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[10],percent=1.0, 
#                                                    train_items=1,user_thres = 2,item_thres=1)

# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[11],percent=1.0, 
#                                                    train_items=1,user_thres = 1,item_thres=1)

# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[12],percent=1.0, 
#                                                    train_items=1,user_thres = 1,item_thres=1)
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[13],percent=1.0, 
#                                                    train_items=1,user_thres = 1,item_thres=1)
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[14],percent=1.0, 
#                                                    train_items=1,user_thres = 2,item_thres=1)
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[15],percent=1.0, 
#                                                    train_items=1,user_thres = 2,item_thres=1)
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[16],percent=1.0, 
#                                                    train_items=1,user_thres = 1,item_thres=1)
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[17],percent=1.0, 
#                                                    train_items=1,user_thres = 1,item_thres=1)
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[18],percent=1.0, 
#                                                    train_items=1,user_thres = 1,item_thres=1)
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[19],percent=1.0, 
#                                                    train_items=1,user_thres = 1,item_thres=1)
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[20],percent=1.0, 
#                                                    train_items=1,user_thres = 1,item_thres=1)
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[21],percent=1.0, 
#                                                    train_items=1,user_thres = 1,item_thres=1)
# targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[22],percent=1.0, 
#                                                    train_items=1,user_thres = 2,item_thres=1)
targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[23],percent=1.0, 
                                                   train_items=1,user_thres = 1,item_thres=1)

find the data/amazon/ratings_Musical_Instruments.csv from data/amazon/ratings_Amazon_Instant_Video
starting dump the id of users and items
Target Statistic: 284 932
0.0035211267605633804
0.09123032444587215


In [65]:
# source target ID pair
targetMetaIdx = 23
sourceMetaIdx = 24
sourceMetaName = preprocess.pathMetaN2MetaN(metaName[sourceMetaIdx],isFile=False)
targetMetaName = metaName[targetMetaIdx] + '_from_' + sourceMetaName
stCommon = preprocess.commonUserIDpair(metaName[sourceMetaIdx], targetMetaName)
print(len(stCommon))

284


### Precessing information

#### ratings_Automotive
find the data/amazon/ratings_Automotive.csv from data/amazon/ratings_Amazon_Instant_Video
starting dump the id of users and items
Target Statistic: 416 1179
0.002403846153846154
0.13363315130099582

32 1024 0.001 5 20000 0.001
Epoch 1233 training loss 1.107381
[0.0986255069851284, 0.14681413048920935, 0.17734715337239002, 0.20165489960442648, 0.21747784287216476, 0.018928110420526743, 0.02180483105459752, 0.022819656296604596, 0.023335650780588593, 0.023612543796413737]
ratings_Amazon_Instant_Video: recall_20 0.071013 recall_40 0.125720 recall_60 0.155027 recall_80 0.184862 recall_100 0.208393
ratings_Amazon_Instant_Video: map_20 0.011142 map_40 0.014171 map_60 0.015170 map_80 0.015931 map_100 0.016369
Best Recall for ratings_Amazon_Instant_Video: recall_20 0.098626 recall_40 0.146814 recall_60 0.177347 recall_80 0.201655 recall_100 0.217478
Best MAP for ratings_Amazon_Instant_Video: map_20 0.018928 map_40 0.021805 map_60 0.022820 map_80 0.023336 map_100 0.023613

find the data/amazon/ratings_Movies_and_TV.csv from data/amazon/ratings_Amazon_Instant_Video
starting dump the id of users and items
Target Statistic: 431 5930
0.002320185614849188
0.13845165435271442

#### ratings_Grocery_and_Gourmet_Food
find the data/amazon/ratings_Grocery_and_Gourmet_Food.csv from data/amazon/ratings_Amazon_Instant_Video
starting dump the id of users and items
Target Statistic: 748 4593
0.001336898395721925
0.24028268551236748

Best Recall for ratings_Amazon_Instant_Video: recall_20 0.085346 recall_40 0.139104 recall_60 0.173789 recall_80 0.196406 recall_100 0.220483
Best MAP for ratings_Amazon_Instant_Video: map_20 0.019637 map_40 0.022741 map_60 0.023715 map_80 0.024228 map_100 0.024616

#### ratings_Patio_Lawn_and_Garden
find the data/amazon/ratings_Patio_Lawn_and_Garden.csv from data/amazon/ratings_Amazon_Instant_Video
starting dump the id of users and items
Target Statistic: 460 1236
0.002173913043478261
0.1477674269193704

32 1024 0.001 5 20000 0.001
Epoch 15037 training loss 0.137601
[0.0945070352010415, 0.14718967502879174, 0.17698162335386314, 0.19658129287466836, 0.21863452005407844, 0.024812326211716765, 0.026940817517639912, 0.02875654073749163, 0.02926778050692659, 0.029525957638299953]
ratings_Amazon_Instant_Video: recall_20 0.022549 recall_40 0.044946 recall_60 0.063541 recall_80 0.077736 recall_100 0.091143
ratings_Amazon_Instant_Video: map_20 0.003400 map_40 0.004354 map_60 0.004944 map_80 0.005250 map_100 0.005488
Best Recall for ratings_Amazon_Instant_Video: recall_20 0.094507 recall_40 0.147190 recall_60 0.176982 recall_80 0.196581 recall_100 0.218635
Best MAP for ratings_Amazon_Instant_Video: map_20 0.024812 map_40 0.026941 map_60 0.028757 map_80 0.029268 map_100 0.029526

#### ratings_Baby
find the data/amazon/ratings_Baby.csv from data/amazon/ratings_Amazon_Instant_Video
starting dump the id of users and items
Target Statistic: 263 737
0.0038022813688212928
0.08448442017346611

32 1024 0.001 5 20000 0.001
Epoch 809 training loss 1.376249
[0.09265434880576835, 0.14306995142957285, 0.17494241650393075, 0.19808847829352566, 0.21434079415151977, 0.021213064177167448, 0.021845956154665284, 0.02228836502207815, 0.02268457928242272, 0.02302752277042695]
ratings_Amazon_Instant_Video: recall_20 0.085369 recall_40 0.132108 recall_60 0.158050 recall_80 0.180343 recall_100 0.196356
ratings_Amazon_Instant_Video: map_20 0.013933 map_40 0.016361 map_60 0.017246 map_80 0.017717 map_100 0.018045
Best Recall for ratings_Amazon_Instant_Video: recall_20 0.092654 recall_40 0.143070 recall_60 0.174942 recall_80 0.198088 recall_100 0.214341
Best MAP for ratings_Amazon_Instant_Video: map_20 0.021213 map_40 0.021846 map_60 0.022288 map_80 0.022685 map_100 0.023028

#### ratings_Digital_Music
targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[22],percent=1.0, 
                                                   train_items=1,user_thres = 1,item_thres=1)
find the data/amazon/ratings_Digital_Music.csv from data/amazon/ratings_Amazon_Instant_Video
starting dump the id of users and items
Target Statistic: 720 4262
0.001388888888888889
0.23128814648249277
32 1024 0.001 5 20000 0.001

Epoch 1008 training loss 1.247612
[0.08987531921285861, 0.14300861248810776, 0.1766286114866557, 0.20407090280907325, 0.22084272194682306, 0.017037482499815616, 0.019295705357660554, 0.020563820039416435, 0.021191521703508805, 0.021521434738171533]
ratings_Amazon_Instant_Video: recall_20 0.075845 recall_40 0.128096 recall_60 0.158355 recall_80 0.180414 recall_100 0.195282
ratings_Amazon_Instant_Video: map_20 0.012432 map_40 0.015393 map_60 0.016425 map_80 0.016956 map_100 0.017217
Best Recall for ratings_Amazon_Instant_Video: recall_20 0.089875 recall_40 0.143009 recall_60 0.176629 recall_80 0.204071 recall_100 0.220843
Best MAP for ratings_Amazon_Instant_Video: map_20 0.017037 map_40 0.019296 map_60 0.020564 map_80 0.021192 map_100 0.021521
32 1024 0.001 5 20000 0.001
targetU2Items = preprocess.source2targetUserRating(metaName[24],ratingFileName[22],percent=1.0, 
                                                   train_items=1,user_thres = 2,item_thres=1)
find the data/amazon/ratings_Digital_Music.csv from data/amazon/ratings_Amazon_Instant_Video
starting dump the id of users and items
Target Statistic: 104 341
0.009615384615384616
0.033408287825248954

32 1024 0.001 5 20000 0.001
Epoch 477 training loss 1.454783
[0.08903910670472183, 0.14593535626658655, 0.17144985228581444, 0.19368959991988388, 0.2208890391067048, 0.01962089795890496, 0.02247086913171176, 0.023375174920163472, 0.02390562303916343, 0.024327354663772886]
ratings_Amazon_Instant_Video: recall_20 0.056963 recall_40 0.139083 recall_60 0.165404 recall_80 0.176760 recall_100 0.192296
ratings_Amazon_Instant_Video: map_20 0.010690 map_40 0.015191 map_60 0.016053 map_80 0.016292 map_100 0.016564
Best Recall for ratings_Amazon_Instant_Video: recall_20 0.089039 recall_40 0.145935 recall_60 0.171450 recall_80 0.193690 recall_100 0.220889
Best MAP for ratings_Amazon_Instant_Video: map_20 0.019621 map_40 0.022471 map_60 0.023375 map_80 0.023906 map_100 0.02432

#### ratings_Musical_Instruments
find the data/amazon/ratings_Musical_Instruments.csv from data/amazon/ratings_Amazon_Instant_Video
starting dump the id of users and items
Target Statistic: 284 932
0.0035211267605633804
0.09123032444587215

32 1024 0.001 5 20000 0.001
Epoch 1518 training loss 0.983145
[0.09563366881978866, 0.14448199889840269, 0.17636573030894803, 0.193453006859947, 0.21226403284762912, 0.020065002989820592, 0.022669917625637366, 0.02317918676937087, 0.023717786041067265, 0.024091747540322984]
ratings_Amazon_Instant_Video: recall_20 0.074986 recall_40 0.126053 recall_60 0.143854 recall_80 0.174511 recall_100 0.186761
ratings_Amazon_Instant_Video: map_20 0.013823 map_40 0.016479 map_60 0.016972 map_80 0.017647 map_100 0.017906
Best Recall for ratings_Amazon_Instant_Video: recall_20 0.095634 recall_40 0.144482 recall_60 0.176366 recall_80 0.193453 recall_100 0.212264
Best MAP for ratings_Amazon_Instant_Video: map_20 0.020065 map_40 0.022670 map_60 0.023179 map_80 0.023718 map_100 0.024092

In [ ]:
# id2userFileName = metaName + '_id2user.pickle'
# id2itemFileName = metaName + '_id2item.pickle'
# id2userFileName2 = metaName2 + '_id2user.pickle'
# id2itemFileName2 = metaName2 + '_id2item.pickle'
# id2userFileName3 = metaName3 + '_id2user.pickle'
# id2itemFileName3 = metaName3 + '_id2item.pickle'
# id2userFileName4 = metaName4 + '_id2user.pickle'
# id2itemFileName4 = metaName4 + '_id2item.pickle'
id2userInstantVideo, id2itemInstantVideo = preprocess.loadID2UandI(metaName1) 

id2userVideoGames,id2itemVideoGames = preprocess.loadID2UandI(metaName2)

id2userBooks,id2itemBooks = preprocess.loadID2UandI(metaName3)

id2userMovies_and_TV,id2itemMovies_and_TV = preprocess.loadID2UandI(metaName4)

id2userTools_and_Home_Improvement,id2itemTools_and_Home_Improvement = preprocess.loadID2UandI(metaName5)

id2userHome_and_Kitchen,id2itemHome_and_Kitchen = preprocess.loadID2UandI(metaName6)

id2userAutomotive,id2itemAutomotive = preprocess.loadID2UandI(metaName7)

id2userDigital_Music,id2itemDigital_Music = preprocess.loadID2UandI(metaName8)

print(id2userInstantVideo[0])
print(id2itemInstantVideo[0])
print(id2userVideoGames[0])
print(id2itemVideoGames[0])
print(id2userMovies_and_TV[0])
print(id2itemMovies_and_TV[0])
print(id2userBooks[0])
print(id2itemBooks[0])
print(id2userTools_and_Home_Improvement[0])
print(id2userHome_and_Kitchen[0])
print(id2userAutomotive[0])
print(id2userDigital_Music[0])

In [ ]:
# userInInstantVideo = [u for uid,u in id2userInstantVideo.items() ]
# itemInInstantVideo = [i for iid,i in id2itemInstantVideo.items() ]
# userInVideoGames = [u for uid,u in id2userVideoGames.items() ]
# itemInVideoGames = [i for iid,i in id2itemVideoGames.items() ]
# userInBooks = [u for uid,u in id2userBooks.items() ]
# itemInBooks = [i for iid,i in id2itemBooks.items() ]
# userInMoviesandTV = [u for uid,u in id2userMovies_and_TV.items() ]
# itemInMoviesandTV = [i for iid,i in id2itemMovies_and_TV.items()]
# userInTools_and_Home_Improvement = [u for uid,u in id2userTools_and_Home_Improvement.items() ]
# itemInTools_and_Home_Improvement = [i for iid,i in id2itemTools_and_Home_Improvement.items()]
# userInHome_and_Kitchen = [u for uid,u in id2userHome_and_Kitchen.items() ]
# itemInHome_and_Kitchen = [i for iid,i in id2itemHome_and_Kitchen.items()]
# userInAutomotive = [u for uid,u in id2userAutomotive.items() ]
# itemInAutomotive = [i for iid,i in id2itemAutomotive.items()]
# userInDigital_Music = [u for uid,u in id2userDigital_Music.items() ]
# itemInDigital_Music = [i for iid,i in id2itemDigital_Music.items()]
sameUser12, _ = preprocess.commonUserItem(metaName,metaName2)
# sameUser34, _ = preprocess.commonUserItem(metaName3,metaName4)
print(len(sameUser12))
print(sameUser12)
# print(len(sameItem))

In [ ]:
# commonUserIdxPair = preprocess.commonUserIDpair(metaName5, metaName6)
commonUserIdxPair12 = preprocess.commonUserIDpair(metaName1, metaName2)
# commonUserIdxPair34 = preprocess.commonUserIDpair(metaName3, metaName4)
commonUserIdxPair = commonUserIdxPair12
print(len(commonUserIdxPair))
commonUser_1 = [i for i,j in commonUserIdxPair]
print(commonUser_1)

In [ ]:
# add the user number threshold
u2items = preprocess.processing(ratingFileName, 5, train_items = 10)
# print(u2items)
i2users = {}
for user in u2items:
    itemList = u2items[user]
    for item in itemList:
        if item not in i2users:
            i2users[item] = [user]
        else:
            i2users[item].append(user)
# print(i2users)
i2usersTemp = {i:u for i,u in i2users.items() if len(u)>3}
print(i2usersTemp)

In [20]:
# generate the data for CMF
# preprocess.genCrossMetaName(metaName[3], metaName[24])
preprocess.transformDataSet2CMF(metaName[24], metaName[3])

-----transforming training file----------
-----transforming testing file----------
-----transforming cross train file----------
